src/vs/workbench/contrib/debug/browser/debugANSIHandling.ts

```diff
@@ -199,7 +213,43 @@ export function handleANSIOutput(text: string, linkDetector: LinkDetector): HTML
 			if (colorType === 'background') {
 				colorNumber += 10;
 			}
-			changeColor(colorNumber, colorType);
+			setBasicColor(colorNumber);
+		}
+	}
+
+	/**
+	 * Calculate and set styling for basic bright and dark ANSI color codes. Uses
+	 * theme colors if available. Automatically distinguishes between foreground
+	 * and background colors; does not support color-clearing codes 39 and 49.
+	 * @param styleCode Integer color code on one of the following ranges:
+	 * [30-37, 90-97, 40-47, 100-107]. If not on one of these ranges, will do
+	 * nothing.
+	 */
+	function setBasicColor(styleCode: number): void {
+		const theme = themeService.getTheme();
+		let colorType: 'foreground' | 'background' | undefined;
+		let colorIndex: number | undefined;
+
+		if (styleCode >= 30 && styleCode <= 37) {
+			colorIndex = styleCode - 30;
+			colorType = 'foreground';
+		} else if (styleCode >= 90 && styleCode <= 97) {
+			colorIndex = (styleCode - 90) + 8; // High-intensity (bright)
+			colorType = 'foreground';
+		} else if (styleCode >= 40 && styleCode <= 47) {
+			colorIndex = styleCode - 40;
+			colorType = 'background';
+		} else if (styleCode >= 100 && styleCode <= 107) {
+			colorIndex = (styleCode - 100) + 8; // High-intensity (bright)
+			colorType = 'background';
+		}
+
+		if (colorIndex !== undefined && colorType) {
+			const colorName = ansiColorIdentifiers[colorIndex];
+			const color = theme.getColor(colorName);
```

Comment:
@isidorn I was able to narrow down the issue to this line. At this point, `colorName` is `terminal.ansiBlack` but `theme.getColor(colorName)` returns `undefined`. This only happens in the test environment; when running in VSCode, the correct color value is returned, so I'm pretty sure it's probably the test theme service initialization. I'm not really sure what else I can try at this point.

Reply:
You need to improve the getColor method of the TestThemeService just so you get the tests to pass. You need to return some mock color
Try it out and if it turns out too complex I can look into it

Reply:
Thanks for pointing that out! I ended up instantiating a custom `TestTheme` with the necessary colors; hope that works alright.